### PR TITLE
Cadc 11901 Add session specific stats and expiry time.

### PR DESCRIPTION
**Pull request by yeunga**
Sample output:
```
% curl -E ~/.ssl/cadcproxy.pem  https://rc-uv.canfar.net/skaha/session/a8jw6e1u

{

  "id": "a8jw6e1u",

  "userid": "alinga",

  "image": "images-rc.canfar.net/skaha/desktop:1.0.3",

  "type": "desktop",

  "status": "Running",

  "name": "alinga-skaha-desktop",

  "startTime": "2022-10-27T16:47:06Z",

  "connectURL": "[ARCADE](https://rc-uv.canfar.net/session/desktop/a8jw6e1u/?password=a8jw6e1u&path=session/desktop/a8jw6e1u/)",

  "requestedRAM": "4Gi",

  "requestedCPUCores": "2",

  "requestedGPUCores": "<none>",

  "coresInUse": "1m",

  "ramInUse": "117Mi",

  "expiryTime": "2022-11-10T16:47:06Z"

}
```

### CODE REVIEW

File: skaha/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ FROM cadc-tomcat:latest
 RUN dnf install -y docker
 
 # install kubernetes client
-ADD https://storage.googleapis.com/kubernetes-release/release/v1.19.0/bin/linux/amd64/kubectl /usr/local/bin/kubectl
+ADD https://storage.googleapis.com/kubernetes-release/release/v1.21.6/bin/linux/amd64/kubectl /usr/local/bin/kubectl
```
Comment (brianmajor) [inline skaha/Dockerfile:8]: Thanks for updating the client version.

File: skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java [hunk 2]
```diff
@@ -367,35 +374,30 @@ public void streamPodLogs(String forUserID, String sessionID, OutputStream out)
...
        
         if (StringUtil.hasLength(vncSession)) {
             Session session = constructSession(vncSession.trim());
+            
+            // get expiry time 
+            List<String> sessionExpiryTimeCMD = getSessionExpiryTimeCMD(k8sNamespace, forUserID, sessionID);
+            String sessionExpiryTime = execute(sessionExpiryTimeCMD.toArray(new String[0]));
+            log.debug("Expiry time: " + sessionExpiryTime + " seconds");
+            Instant instant = Instant.parse(session.getStartTime());
+            instant = instant.plus(Integer.parseInt(sessionExpiryTime), ChronoUnit.SECONDS);
```
Comment (brianmajor) [inline skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java:391]: What's the difference between using Instance/ChronoUnit and Date?  Just a newer way of doing it?

**Review by brianmajor — APPROVED** [review]

File: skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java (hunk 2 above)
Comment (yeunga) [reply skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java:391]: Most of the methods in Date have been deprecated. Using Instant here only requires two lines of code, so I used it.

Comment (brianmajor) [reply skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java:391]: Okay, thanks, good to know.

**Pull request merged**